keyboards/abacus/keymaps/default/keymap.c

```diff
@@ -0,0 +1,140 @@
+    UPSIDEDOWN,
+    NOMOUTH,
+    PARTY,
+    HEART,
+    EGGPLANT,
+    PEACH,
+    EMOJI100,
+    EMOJIB
+};
+
+const uint32_t PROGMEM unicode_map[] = {
+    [LOVEEYES]  = 0x1f60d,
+    [THINK] = 0x1f914,
+    [UPSIDEDOWN] = 0x1f643,
+    [NOMOUTH] = 0x1f636,
+    [PARTY]  = 0x1f973,
+    [HEART] = 0x1f495,
+    [EMOJI100] = 0x1f4af,
+    [PEACH] = 0x1f351,
+    [EGGPLANT] = 0x1f346,
+    [EMOJIB] = 0x1f171
+};
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        KC_ESCAPE, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPACE,
+        KC_TAB, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCOLON, KC_BSLASH,
+        KC_LSHIFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMMA, KC_DOT, KC_UP, KC_DELETE,
+        KC_LCTRL, KC_LGUI, MO(_UP_1), KC_SPACE, KC_ENTER, KC_LALT, KC_LEFT, KC_DOWN, KC_RIGHT
+    ),
+    [_UP_1] = LAYOUT(
+        KC_GRAVE, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, _______,
+        KC_CAPS, _______, _______, _______, _______, _______, _______, _______, KC_LBRACKET, KC_RBRACKET, KC_QUOTE, KC_SLASH,
+        _______, _______, _______, _______, _______, _______, _______, _______, KC_MINUS, KC_EQUAL, MO(_UP_2), _______,
+        _______, _______, _______, _______, _______, _______, KC_HOME, _______, KC_END
+    ),
+    [_UP_2] = LAYOUT(
+        NICKURL, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, _______,
+        _______, KC_F11, KC_F12, RGB_MODE_PLAIN, RGB_MODE_BREATHE, RGB_MODE_RAINBOW, RGB_MODE_SWIRL, RGB_MODE_SNAKE, RGB_MODE_KNIGHT, RGB_MODE_GRADIENT, XXXXXXX, RGB_TOG, 
+        _______, X(LOVEEYES), X(THINK), X(UPSIDEDOWN), X(NOMOUTH), X(PARTY), X(PEACH), X(HEART), X(EGGPLANT), X(EMOJI100), X(EMOJIB), _______,
+        _______, _______, _______, _______, _______, _______, RGB_HUI, RGB_SAI, RGB_VAI
+    )
+ 
+ 
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case NICKURL:
+            if (record->event.pressed) {
+                SEND_STRING("https://www.github.com/nickolaij");
+            } else {
+                tap_code(KC_ENTER);
+            }
+            return false;
+
+        default:
+            return true;
+
+    }
+}
+
+
+void dip_switch_update_user(uint8_t index, bool active) { 
+    switch (index) {
+        case 0:
+            if(active) {
+                switch(biton32(layer_state)) {
+                case _BASE:
+                    tap_code16(C(KC_F));
+                    break;
+                case _UP_1:
+                    tap_code(KC_MUTE);
+                    break;
+                case _UP_2:
+                    tap_code(KC_MEDIA_PLAY_PAUSE);
+                    break;
+                }
+        }
+    }
+}
+
+
+void matrix_init_user(void) {
+    set_unicode_input_mode(UC_WINC);
+}
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+
+    switch(biton32(layer_state)) {
```

Comment:
Specifically, this is typedeffed based on the int size that is used for the layer_state bitmask. 

```suggestion
    switch(get_highest_layer(layer_state)) {
```